lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -421,19 +421,28 @@ defmodule ExDoc.Formatter.HTML.Autolink do
   will get translated to the new href of the function.
   """
   def elixir_functions(bin, project_funs, extension \\ ".html", lib_dirs \\ elixir_lib_dirs()) when is_binary(bin) do
-    module_re = Regex.source(~r{(([A-Z][A-Za-z_\d]+)\.)+})
-    fun_re = Regex.source(~r{([ct]:)?(#{module_re}([a-z_]+[A-Za-z_\d]*[\\?\\!]?|[\{\}=&\\|\\.<>~*^@\\+\\%\\!-]+)/\d+)})
-    regex = ~r{(?<!\[)`\s*(#{fun_re})\s*`(?!\])}
-
-    Regex.replace(regex, bin, fn all, match ->
-      {prefix, module, function, arity} = split_function(match)
+    module_re = ~r{(?|(?|[A-Z][A-Za-z_\d]+)\.)*}.source
+    fun_re = ~r{([ct]:)?(#{module_re}([a-z_]+[A-Za-z_\d]*[\\?\\!]?|[\{\}=&\\|\\.<>~*^@\\+\\%\\!-]+)/\d+)}.source
+    normal_link_re = ~r{(?<!\[)`\s*(#{fun_re})\s*`(?!\])}.source
+    custom_link_re = ~r{(?|\[(.*)\])\(`(#{fun_re})`\)}.source
+    re = ~r{#{custom_link_re}|#{normal_link_re}}
```

Comment:
@josevalim instead of making this regex more complicated, we could process custom links in a separate pass, WDYT?

I put together a fresh project to play with this stuff, see how that 2 pass thing could look like: https://github.com/wojtekmach/autolink/blob/master/lib/autolink.ex#L91:L95 In that snippet I'm also unifying `local_doc` and `elixir_functions` and I'm still exploring whether that's a good idea, so let's ignore that for now :)

Reply:
:+1: for two passes!